File: app/src/main/java/com/pratik/moviesapp/controller/MovieCallBack.java
package com.pratik.moviesapp.controller;

import android.support.annotation.NonNull;

import com.pratik.moviesapp.models.Movie;


public interface MovieCallBack {

    void getPopularMovies(@NonNull Movie movie);

    void getTopRateMovies(@NonNull Movie movie);

    void onPopularMoviesError(@NonNull Throwable throwable);

    void onTopRatedMoviesError(@NonNull Throwable t);
}

File: app/src/main/java/com/pratik/moviesapp/controller/MovieCallBack.kt
package com.pratik.moviesapp.controller

import com.pratik.moviesapp.models.Movie

interface MovieCallBack {
    fun getPopularMovies(movie: Movie)
    fun getTopRateMovies(movie: Movie)
    fun onPopularMoviesError(throwable: Throwable)
    fun onTopRatedMoviesError(t: Throwable)
}